torch/_meta_registrations.py

```diff
@@ -6178,6 +6178,34 @@ def meta_searchsorted(
         return torch.empty((), dtype=dtype, device=sorted_sequence.device)
 
 
+def _check_for_unsupported_isin_dtype(dtype):
+    torch._check(
+        not any(
+            dtype == ud
+            for ud in [torch.bool, torch.bfloat16, torch.complex128, torch.complex64]
+        ),
```

Comment:
Aren't those equivalent?
```suggestion
            dtype not in (torch.bool, torch.bfloat16, torch.complex128, torch.complex64),
```